### PR TITLE
visual c++ doesn't have C11 atomics

### PR DESCRIPTION
**Pull request by xenu**

### CODE REVIEW

File: src/refcount.h [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
+#else
+/* Visual C++ doesn't support C11 atomics yet. However, Windows documentation
+ * guarantees that simple reads and writes are atomic:
+ * https://docs.microsoft.com/en-us/windows/win32/sync/interlocked-variable-access
+ */
+#  include <windows.h>
+
+typedef volatile size_t Refcount;
+
+#  define refcount_inited(counter) (*(counter) != 0)
```
Comment (Leont) [inline src/refcount.h:21]: Wouldn't using counter as a pointer require it to be allocated first? This looks weird to me.

**Comment by Leont** [comment]
Yeah, atomics support on other/older compilers is a known todo list item. I'm going to have to find a solution for older gcc/clang too.

File: src/refcount.h (hunk 1 above)
Comment (xenu) [reply src/refcount.h:21]: This is exactly the same logic as in the C11 version. The code calls refcount_inited like this:
```c
if (!refcount_inited(&thread_counter)) {
```

Comment (Leont) [reply src/refcount.h:21]: Right, I forgot about that.

**Pull request merged**